### PR TITLE
release: v1.36.0

### PR DESCRIPTION
**Pull request by abernix**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it.
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
    - Does this PR target the right branch? (usually, `main`)
    - Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
    - If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_lrlna_graphql_validation_both.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+### Use both Rust and JS Validation as default([issue#4159](https://github.com/apollographql/router/issues/4159))
+
+As part of the process to replace JavaScript validation with a more performant Rust validation in the Router, we are enabling the Router to run both validations as a default. This allows us to definitively assess reliability and stability of Rust validation before completely removing JavaScript validation. As before, it's possible to toggle between implementations using `experimental_graphql_validation_mode` config key. Possible values are: `new` (runs only Rust-based validation), `legacy` (runs only JS-based validation), `both` (runs both in comparison, logging errors if a difference arises).
```
Comment (shorgi) [inline .changesets/feat_lrlna_graphql_validation_both.md:3]:
```suggestion
### Use both Rust and JS Validation as default([issue#4159](https://github.com/apollographql/router/issues/4159))

As part of the process to replace JavaScript validation with a more performant Rust validation in the router, we are enabling the router to run both validations as a default. This allows us to definitively assess reliability and stability of Rust validation before completely removing JavaScript validation. As before, it's possible to toggle between implementations using the `experimental_graphql_validation_mode` config key. Possible values are: `new` (runs only Rust-based validation), `legacy` (runs only JS-based validation), `both` (runs both in comparison, logging errors if a difference arises).
```

File: .changesets/fix_bryn_span_mode.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+`telemetry.instrumentation.spans.span_mode` was not being correctly applied, resulting in extra request spans that should not have been present in spec compliant mode.
+
+In spec compliant mode the spans should have the hierarchy:
+
+`router.supergraph.subgraph`
+
+but were instead output as:
+
+`request.router.supergraph.subgraph`
```
Comment (shorgi) [inline .changesets/fix_bryn_span_mode.md:11]:
```suggestion
### `span_mode: spec_compliant` not applied correctly ([Issue #4335](https://github.com/apollographql/router/issues/4335))

Previously, `telemetry.instrumentation.spans.span_mode.spec_compliant` was not being correctly applied. This resulted in extra request spans that should not have been present in spec compliant mode, where `router.supergraph.subgraph` was incorrectly output as `request.router.supergraph.subgraph`. This has been fixed in this release.

```

**Review by shorgi — COMMENTED** [review]
changelog copyedits

File: .changesets/fix_bryn_span_mode.md (hunk 2 above)
Comment (abernix) [reply .changesets/fix_bryn_span_mode.md:11]: love this.  thank you!

**Review by lennyburdette — APPROVED** [review]

**Review by lrlna — APPROVED** [review]

**Pull request merged**